Add pokeball command to actions

PixeLInc opened the pull request:
Adds the remaining Pokeball command action for [Card Link](https://github.com/LeggoMyEcho/DrawBot/projects/1#card-33733936)

It provides a 3/7 chance to catch someone in your Pokeball.

PixeLInc committed:
Add pokeball command to actions

z64 commented on src/draw_bot/handlers/action.cr:
```diff
@@ -48,4 +48,23 @@ module DrawBot
+
+    client.create_message(
+      payload.channel_id,
+      "#{payload.author.username} throws a pokeball at #{user}, #{payload.author.username} #{result} #{user}"
```
I think we should retain `display_name` behavior:

```cr
author_name = payload.member.try(&.nick) || payload.author.name
```

We can look into making this more ergonomic sometime (here or upstream), but this is fine for now.

z64 commented on src/draw_bot/handlers/action.cr:
```diff
@@ -48,4 +48,23 @@ module DrawBot
+    SplitParser.new("~pokeball", join_after: 1, min_args: 1)
+  ) do |payload, context|
+    args = context[DrawBot::SplitParser::Results].arguments
+    user = args[0]
```
Strictly speaking, the input text can be anything. In these style of commands, I'd like to call the argument `subject` instead of `user` - since it's just a `String` that we have no expectations of.

If we actually did something like parsed the string into a `User`, that would be a different story.

PixeLInc committed:
Resolve member nick, change to subject

z64 committed:
Consolidate some code

I think these are acceptable changes without losing much readability,
since the commands are so short.

z64 commented:
I pushed a few nitpicks to save the back/forth. I'm good with this if it looks fine to you.
